src/writer.js

```diff
@@ -0,0 +1,377 @@
+let faceUpCard1;
+var isFirstClick = true;
+let isLoadedGame = false;
+export const callback = function(mutationsList, observer) {
+  document.getElementsByClassName('gameField')[0].onclick = function (event) {
+    let target = event.target;
+    if(!target.classList.contains('card')) return;
+    toggleCard(target);
+
+    if (isFirstClick) {
+      isFirstClick = false;
+      startStop();
+    }
+
+    if (!target.classList.contains('cardBack')) {
+      faceUpCounter += 1;
+      target.classList.add('noclick');
+      //if two cards are faced up
+      if (faceUpCounter >= 2) {
+        document.getElementsByClassName('gameField')[0].classList.add('noclick');
+        setTimeout(() => {document.getElementsByClassName('gameField')[0].classList.remove('noclick')}, 200);
+        if (!isLoadedGame) {
+          document.getElementsByClassName('turnsCounter')[0].value++;
+        }
+        compareCards(target, faceUpCard1);
+
+        let cards = document.getElementsByClassName('card');
+        if (emptyCardNumber == classArrLength) {
+          let time = document.getElementsByClassName('stopwatch')[0].value;
+          let turns = document.getElementsByClassName('turnsCounter')[0].value;
+
+          let fieldSize = document.getElementsByClassName('fieldSelector')[0].value;
+          let result = {
+            'time': time,
+            'turns': turns,
+            'size': fieldSize,
+          }
+
+          if (isLoadedGame) {
+            setTimeout(function () {
+              showMsg(`<b>Well... <br> You'd better play without saving and loading... <br></b> Statistics are nor recorded for loaded games`);
+              document.getElementsByClassName('msgDiv')[0].append(createHtml('button', 'msgButton', 'Play Again'));
+              document.getElementsByClassName('msgButton')[0].addEventListener('click', changeFieldSize);
+            }, 300);
+          }
+
+          //change HTML and localStorage if new result is better
+          if (isFirstBetter(result, localStorage.getItem(result.size))) {
+            setTimeout(function () {
+              showMsg(`<b>Congratulations! <br> New Record! <br></b>You paired all the cards in ${time} and ${turns} turns <br>`);
+              document.getElementsByClassName('msgDiv')[0].append(createHtml('button', 'msgButton', 'Play Again'));
+              document.getElementsByClassName('msgButton')[0].addEventListener('click', changeFieldSize);
+            }, 300);
+            changeLeadResult(result);
+          }
+          else {
+            setTimeout(function () {
+              showMsg(`<b>Good, But You Gotta Try Harder To Beat The Record! <br> </b>You paired all the cards in ${time} and ${turns} turns <br>`);
+              document.getElementsByClassName('msgDiv')[0].append(createHtml('button', 'msgButton', 'Play Again'));
+              document.getElementsByClassName('msgButton')[0].addEventListener('click', changeFieldSize);
+            }, 300);
+          }
+
```

Comment:
it looks like 3 same pieces of code. can it be wrapped into function?

Reply:
Done